added test setup and test

iswanj opened the pull request:
@vutran Added Jest and test too,  But I didn't mock the `got` module. If you want do that let me know I'll do that too.

Commit:
added test setup and test

vutran commented on __tests__/index-test.js:
```diff
@@ -0,0 +1,24 @@
+require("babel-polyfill");
```
I don't think we need to add this here.

vutran commented on __tests__/index-test.js:
```diff
@@ -0,0 +1,24 @@
+
+describe('Hacker News Plugin', () => {
+  it('execute returns new 10 items', async () => {
+    const hack = require('../index.js');
```
Let's hoist these `require()`'s to the top, we don't need to `require()` for each spec.

vutran commented on package.json:
```diff
@@ -17,11 +17,17 @@
   "scripts": {
-    "lint": "eslint index.js"
+    "lint": "eslint index.js",
+    "test": "jest --verbose --watch"
```
Let's use `jest --coverage` instead since we'll be integrating code coverage in the future.

You can create a new command `test-watch` for `--watch`

vutran reviewed (changes requested):
Can you mock the data that is being requested? You can refer to these [instructions](https://github.com/vutran/dext-github-plugin/issues/6#issuecomment-250889634) for more info. Also, the official jest [mocks doc](https://facebook.github.io/jest/docs/manual-mocks.html).

vutran commented:
Aside from those, these look great! 👍

Commit:
Added got module mock and fixed commented issues

iswanj commented:
@vutran Fixed commented issues and mock function. But I have no idea about the data format that server sent.
I sent this, 

```
{
    title: 'I Passed on Tesla: VCs Confess Their Biggest Regrets',
    subtitle: 'Udo_Schmitz',
    arg: 'http://fortune.com/2016/10/06/uber-apple-snapchat-missed-deals/',
    icon: { path: './icon.png' }
}
```

but the result 

```
{ 
    title: undefined,
    subtitle: undefined,
    arg: undefined,
    icon: { path: './icon.png' }
}
``

BTW, in a test file, I'm checking the length of the result, so this does not affect the test. 
```

vutran commented:
See [this url](https://hacker-news.firebaseio.com/v0/item/12659141.json) for example.

``` json
{
  "by": "Anon84",
  "descendants": 22,
  "id": 12659141,
  "kids": [
    12660370,
    12659509,
    12660390,
    12659406,
    12659486
  ],
  "score": 143,
  "time": 1475830860,
  "title": "Foundations of Data Science [pdf]",
  "type": "story",
  "url": "https://www.cs.cornell.edu/jeh/book2016June9.pdf"
}
```

Commit:
added more data and make sure its only send 10 items

Commit:
fixed undefinde result issue

iswanj commented:
@vutran Can you check the code again :+1:

vutran committed:
updated mocks and tests

- renamed test file to index.js
- simplified mock got() function
- rename mock data vars
- move import to top in test file
- use toBeUndefined()
- new spec for invalid command

iswanj committed:
Merge pull request #1 from vutran/iswanj-add-test

Iswanj add test

vutran commented:
@iswanj Thanks for the great work! #happyhacktoberfest

iswanj commented:
:smile: Thank you too, I've learned few things from you.
